tethys_cli/install_commands.py

```diff
@@ -869,9 +869,10 @@ def validate_schema(check_str, check_list):
     )
 
 
-def successful_exit(app_name, action="installed"):
+def successful_exit(app_name, action="installed", do_exit=True):
```

Comment:
If you look at old line 874 - and not the name of the function - `successful_exit` would always exit. That's why I made the do_exit default to True, so that there's not any unintended new behavior from pre-existing code. However, I now see the silliness of having a function called `successful_exit` that has the exit as optional via `do_exit` is odd. I'm exploring now doing it the way you suggest, but rather than just pull `exit(0)` out, I'm just completely getting rid of the `successful_exit` function and just calling `write_success` everywhere `successful_exit` was called. But upon looking at it more, I'm curious why we even need `exit(0)` instead of just a `return` since everywhere it is called is inside of a function. Am I missing something there?

Reply:
We discussed this in a meeting and decided that I can replace `exit(0)` with `return` as long as the tests are updated to ensure they are actually exiting in the expected place. I have now done this.